src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,110 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final double INCREASE_CAPACITY = 1.5;
+    private T[] elements;
+    private int size;
+
+    public ArrayList() {
+        elements = (T[]) new Object[DEFAULT_CAPACITY];
+    }
+
+    private void ensureCapacity() {
+        if (size() == elements.length) {
+            int newCapacity = (int) (elements.length * INCREASE_CAPACITY);
+            T[] newElements = (T[]) new Object[newCapacity];
+            System.arraycopy(elements, 0, newElements, 0, size);
+            elements = newElements;
+        }
```

Comment:
The ensureCapacity method could be simplified. There's no need to call size() since you can directly access the size field. Also, the resizing logic should be moved to a separate method according to the Single Responsibility principle.